### ISSUE TITLE
Sync patterns across windows

### ISSUE THREAD

**Issue by gkinsman** [jdhayford/acid-tabs-extension#5]
Hi there! Just started using this extension and loving it so far.

It would be amazing if there was an option in the config for each pattern to keep that group to its own window. For example, enabling it for 'mail' would move and keep the mail group to its own window.

Would a simple checkbox in the UI work, or could this feature be generalised a bit? Could be you want to group tab groups by some identifier which would keep them to the same window?

The use case for this is so that I could keep github tabs to one window, mail to another etc.

Thoughts? Is this even possible with the chrome API?

**Comment by jdhayford** [comment]
Hey @gkinsman, I definitely understand and like the idea here. I believe I had actually originally built it in a way that only had one group across your chrome windows, but then found the experience got pretty weird if you opened a new tab in a different window and it either disappeared to the other one or switched the focus over. Making it group per windows helped that, buuuuut I also do understand where it would be nice to do this.

In terms of how we could introduce it, I would love to keep each rule as simple as possible for as long as possible so I would prefer some sort of high level setting. A simple approach might be to have a global setting to either do per window groups or groups across all. The identifier (I'm assuming you mean some pattern in the group name) would be a solid approach as well, if customizable by the user, to give more flexibility at the cost of simplicity.

So in summary, I definitely think this is possible given the Chrome API. My personal preference would be to make it a global setting (I'm working on adding a settings page for things like this to live soon) and maybe with some way for the user to have a per rule way of driving the behavior.

**Comment by gkinsman** [comment]
Ah that's an interesting UX issue, I hadn't thought of that. Perhaps a way around it might be, if possible, to replace the moved tab with a tombstone saying 'You moved this tab' and a big button saying 'cool' that dismisses the tab. There might be better ways too.

I think moving to a system where you declare your window setup like this would be a dramatic change to workflow, as you're forcing people to start using the tabs they already have open. I think this is a good thing, but maybe there might be a way to support both workflows. What if you could open a new window and somehow say its an ad-hoc session that wouldn't group your tabs? Then maybe pressing the plugin button would send them to their respective groups in whatever window.

I'm just spitballing a bit here, but maybe the issue is less about the setting and more how to get the UX right to make it easy to use. Maybe introducing this global setting might be enough for now, and then once people are using it the UX could be refined?

**Comment by yegle** [comment]
I'm fine being able to have the same group in different window. But I'd like to have a way to "merge group A from all windows to the group in this window".
